### PR TITLE
[WIP] Improve CommitStatus

### LINKED ISSUE #25990: Pull request check list is limited to 30 entries
### Description

This is linked to the DEFAULT_PAGING_NUM value.
The computation of the check summary status is also wrong due to this limit: A status can be in progress and the summary value having the value "All checks were successful"

### Gitea Version

1.19

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Docker

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by yp05327**
Mentioned in https://github.com/go-gitea/gitea/issues/25990#issuecomment-1653023772
Fix #25990

Changes:
1. Using `Int` to save CommitStatus in DB
2. Using `min` in sql to get the CommitStatus instead of calculate CommitStatus in Gitea side
3. Adding a pager bar in web ui

### CODE REVIEW

File: templates/repo/commit_status.tmpl [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-{{end}}
-{{if eq .State "success"}}
-	{{svg "octicon-check" 18 "commit-status icon text green"}}
-{{end}}
-{{if eq .State "error"}}
+{{if eq .State 1}}
 	{{svg "gitea-exclamation" 18 "commit-status icon text red"}}
 {{end}}
-{{if eq .State "failure"}}
+{{if eq .State 2}}
```
Comment (CaiCandong) [inline templates/repo/commit_status.tmpl]: I believe that using direct numbers for judgment here leads to poor code readability and maintainability. My opinion is to continue using the previous code on the frontend page.When passing the state to the frontend, perform a conversion from numbers to strings.

File: modules/structs/commit_status.go [hunk 2]
```diff
@@ -5,39 +5,34 @@ package structs
...
+}
+
+// String returns the string name of the CommitStatusState
+func (css CommitStatusState) String() string {
+	return commitStatusNames[css]
 }
 
 // NoBetterThan returns true if this State is no better than the given State
-// This function only handles the states defined in CommitStatusPriorities
 func (css CommitStatusState) NoBetterThan(css2 CommitStatusState) bool {
```
Comment (CaiCandong) [inline modules/structs/commit_status.go:40]: This requires checking for special values, enhancing the robustness of the function, and avoiding issues caused by incorrect usage by the caller, as seen in #26121.

File: templates/repo/commit_status.tmpl (hunk 1 above)
Comment (yp05327) [reply templates/repo/commit_status.tmpl]: Ah, forgot that we have CommitStatusState.IsXXXX.
I think this is the best solution.

File: modules/structs/commit_status.go (hunk 2 above)
Comment (yp05327) [reply modules/structs/commit_status.go:40]: Finally, I want to remove this function. 
It is only used in `MergeRequiredContextsCommitStatus`. But have no good idea about how to rewrite it now.
Or maybe I will add `CommitStatusState.IsValid` which should be called before we compare them.

**Pull request closed without merging**